WcaOnRails/app/models/competition.rb

```diff
@@ -1771,7 +1771,8 @@ def update_persons_wcif!(wcif_persons, current_user)
     wcif_persons.each do |wcif_person|
       local_assignments = []
       registration = registrations.find { |reg| reg.user_id == wcif_person["wcaUserId"] }
-      # If no registration is found, assume that this is a non-competing staff member being added.
+      # If no registration is found, and the Registration is marked as non-competing, add this person as a non-competing staff member.
+      next unless registration || !wcif_person["registration"] || wcif_person["registration"]["is_competing"] != false
```

Comment:
First part of the OR is obvious.

Third part should use `wcif_person["registration"]["isCompeting"]` in camel case instead of `is_competing` snake case. Also, at this point in time the WCIF has already been schema-checked (this is actually the first thing that happens when hitting the PATCH endpoint) so I believe it is fair to assume that `isCompeting` is always a boolean anyways. So does explicitly checking `!= false` make sense?!

In a wider sense, what about reworking this into `next if !registration.present? && wcif_person["registration"] && !wcif_person["registration"]["isCompeting"]`? I find it much easier to understand than the implicitly negated form baked into the `unless`, although that might just be personal preference. Totally open to debate and to being convinced of the current `unless` version.

Reply:
Hold on, now I confused myself with that suggestion about reworking the `next` condition. What are we even trying to achieve here? Sorry :see_no_evil: :woozy_face:

Reply:
Oops, fixing to isCompeting.

What are we trying to achieve? The idea was to go back to the old behavior (see base side of https://github.com/thewca/worldcubeassociation.org/pull/7601/files) unless the app PATCHing explicitly says isCompeting: false. This means all the implicitly-added organizer entities won't show up as non-competing registrations anymore.

I was trying to explicitly handle a missing isCompeting differently than a set-to-false isCompeting (does the schema validation fail if isCompeting is missing?) My assumption was that missing isCompeting meant the app writer didn't consider that field, and set-to-false means they considered it and said "yes this person is not competing", but both nil and false would be treated as false. Thoughts?

Agree with reversing the negatives.

Reply:
The idea makes sense. So what about a two-step approach?
```ruby
# ...from before, nothing new:
registration = registrations.find { |reg| reg.user_id == wcif_person["wcaUserId"] }

adding_non_competing = wcif_person["registration"].present? && wcif_person["registration"]["is_competing"] == false
registration ||= ...whatever... if adding_non_competing

# similar to the old status quo from the PR you referenced, but with a tiny bit more readable `.present?` now
next unless registration.present?
```

Reply:
OK, done! Rubocop doesn't like `registration ||= ... if adding_non_competing`, so I went with a normal if statement to appease it.